vertx-config/src/main/java/io/vertx/config/impl/spi/HttpConfigStore.java

```diff
@@ -32,21 +36,30 @@
  */
 public class HttpConfigStore implements ConfigStore {
 
-  private final String host;
-  private final int port;
-  private final String path;
   private final HttpClient client;
+  private final RequestOptions requestOptions;
 
-  public HttpConfigStore(String host, int port, String path, HttpClient client) {
-    this.host = host;
-    this.port = port;
-    this.path = path;
-    this.client = client;
+  public HttpConfigStore(Vertx vertx, JsonObject configuration) {
+    String host = configuration.getString("host");
```

Comment:
I think instead we should not extract each piece of information from the configuration but we use a specific json object that is unmarshalled to a RequestOptions using the JsonObject constructor ?

Reply:
I was thinking of that may break the configuration format, but yes, we can do that by making it compatible with old format.

I was also thinking of that together with the `SpringConfigServerStore` which is actually a HTTP request as well. So maybe we can unify the configuration format and making `SpringConfigServerStore` extends `HttpConfigStore` ?

Reply:
I prefer to not have this coupling between stores.

You are right for format, it's likely better to keep it like this.